src/osp/OneStepProverHostIo.sol

```diff
@@ -29,6 +30,15 @@ contract OneStepProverHostIo is IOneStepProver {
     uint64 private constant INBOX_HEADER_LEN = 40;
     uint64 private constant DELAYED_HEADER_LEN = 112 + 1;
 
+    ICustomDAProofValidator public customDAValidator;
+
+    function setCustomDAValidator(
+        ICustomDAProofValidator _validator
+    ) external {
+        // TODO: Add appropriate access control
+        customDAValidator = _validator;
+    }
+
```

Comment:
we don't keep any storage in the OSP contract, so instead I think we have 2 options
1. store the var in the rollup and have the OSP read from the rollup
2. make this immutable and set in the constructor, but we have to consider how it works with the rollup creator

Reply:
I would prefer to make this immutable because this allow more deterministic behavior, otherwise the rollup owner may change the DA contract and make a assertion invalid.